LibSWBF2.NET/API/APIWrapper.cs

```diff
@@ -16,44 +16,44 @@ internal static class APIWrapper
 
         // Logging //
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern void LOG_SetCallbackMethod(LogCallback Callback);
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern void LOG_SetLogfileLevel(Logging.ELogType LogfileLevel);
 
 
         // Level //
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern IntPtr Level_FromFile([MarshalAs(UnmanagedType.LPStr)] string path);
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern void Level_Destroy(IntPtr level);
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         [return: MarshalAs(UnmanagedType.U1)]
         public static extern bool Level_IsWorldLevel(IntPtr level);
         
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern void Level_GetModels(IntPtr level, out IntPtr modelArr, out uint modelCount);
 
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern IntPtr Level_GetModel(IntPtr level, [MarshalAs(UnmanagedType.LPStr)] string modelName);
 
         // Model //
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         [return: MarshalAs(UnmanagedType.LPStr)]
         public static extern string Model_GetName(IntPtr model);
         
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         public static extern void Model_GetSegments(IntPtr model, out IntPtr segmentArr, out uint segmentCount);
-		
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         [return: MarshalAs(UnmanagedType.U1)]
         public static extern bool Model_IsSkeletalMesh(IntPtr model);
         
-        [DllImport("LibSWBF2", CallingConvention = CallingConvention.Cdecl)]
+        [DllImport("SWBF2", CallingConvention = CallingConvention.Cdecl)]
         [return: MarshalAs(UnmanagedType.U1)]
```

Comment:
This way, on Mac and Linux Mono automatically searches for libSWBF2.dylib and libSWBF2.so, and just SWBF2.dll on Windows.  Hence the LibSWBF2.dll -> SWBF2.dll name change in the .csproj PostBuildEvent for Windows.

Reply:
I'd _really really_ like to avoid renaming the DLL. We should declare preprocessor symbols for the different platforms and switch accordingly.

Reply:
Sounds good!